src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/DiscardSyntaxClassifier.cs

```diff
@@ -0,0 +1,125 @@
+        {
+            if (syntax.IsKind(SyntaxKind.DiscardDesignation) || syntax.IsKind(SyntaxKind.DiscardPattern))
+            {
+                result.Add(new ClassifiedSpan(syntax.Span, ClassificationTypeNames.Keyword));
+                return;
+            }
+
+            var _ =
+                TryClassifyCallArgument(syntax, semanticModel, result, cancellationToken) ||
+                TryClassifyLambdaParameter(syntax, semanticModel, result, cancellationToken) ||
+                TryClassifySymbol(syntax, semanticModel, result, cancellationToken);
+        }
+
+        private bool TryClassifyCallArgument(SyntaxNode syntax, SemanticModel semanticModel, ArrayBuilder<ClassifiedSpan> result, CancellationToken cancellationToken)
+        {
+            if (!(syntax is ArgumentSyntax argument))
+            {
+                return false;
+            }
+
+            // in arguments we don't need to go on if the out is missing or the argument isn't inside a tuple
+            if (!argument.RefOrOutKeyword.IsKind(SyntaxKind.OutKeyword) &&
+                !argument.IsParentKind(SyntaxKind.TupleExpression))
+            {
+                return false;
+            }
+
+            var operation = semanticModel.GetOperation(argument.Expression, cancellationToken);
+            if (!(operation is IDiscardOperation discardOperation))
+            {
+                return false;
+            }
+
+            switch (discardOperation.Syntax)
+            {
+                case DeclarationExpressionSyntax decl:
+                    result.Add(new ClassifiedSpan(decl.Designation.Span, ClassificationTypeNames.Keyword));
+                    return true;
+                case IdentifierNameSyntax discard:
+                    result.Add(new ClassifiedSpan(discard.Span, ClassificationTypeNames.Keyword));
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        private bool TryClassifyLambdaParameter(SyntaxNode syntax, SemanticModel semanticModel, ArrayBuilder<ClassifiedSpan> result, CancellationToken cancellationToken)
+        {
+            if (!(syntax is LambdaExpressionSyntax lambda))
+            {
+                return false;
+            }
+
+            var symbolInfo = semanticModel.GetSymbolInfo(lambda, cancellationToken);
+            var symbol = symbolInfo.Symbol as IMethodSymbol;
+
+            if (symbol == null)
+            {
+                return false;
+            }
+
+            DataFlowAnalysis? _lazyAnalysis = null;
+
+            foreach (var parameter in symbol.Parameters)
+            {
+                if (parameter.IsDiscard)
+                {
+                    result.Add(new ClassifiedSpan(parameter.Locations[0].SourceSpan, ClassificationTypeNames.Keyword));
+                }
+                else if (parameter.Name == "_") //unfortunately parameter.IsDiscard currently doesn't work for all cases
```

Comment:
Is this a bug with the compiler or is this an unused parameter that looks like a discard but is actually in scope?

Reply:
This is hit by the `DiscardInLambda` test which looks like this:
```cs
class X
{
    void N()
    {
        System.Func<int, int> a = (int _) => 0;
    }
}
```

The `IsDiscard` property for the symbol is false which is why i do a flow analysis in the body of the if, to check weather or not the parameter is actually used or not, in which case it is classified.

Unfortunately i lack the experience to confidently decide if that is a bug or not but as a user i would expect `IsDiscard` to return true in the case where the parameter isn't used in the lambda

Reply:
Judging by [this section](https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs#L198-L201) in the lambda binder the behavior seems to be intentional as only a single parameter is given. 

https://github.com/dotnet/csharplang/issues/111 makes it clear that this is intentional since otherwise it would break backwards compatibility. For purposes of classification i think it would still be nice to treat a single unused `_` as a discard and highlighting it as a keyword. 

However i think that changing the implementation of `IsDiscard` might break consumers. Currently consumers inside the Roslyn codebase are mostly tests, as well as the [`AbstractSymbolDisplayService`](https://github.com/dotnet/roslyn/blob/master/src/Features/Core/Portable/LanguageServices/SymbolDisplayService/AbstractSymbolDisplayService.AbstractSymbolDescriptionBuilder.cs#L615). These consumers might not even need a change however `IParameterSymbol.IsDiscard` is part of the public API since 4dea44169d812238c2c751ab342b038106a3b537.

I think therefore there are 3 options:
1. change the lambda binder so `IsDiscard` for single unsued `_` becomes true
2. keep the binder as it is and check this special case in the `DiscardSyntaxClassifier`
3. remove this edge case from `DiscardSyntaxClassifier`, not highlighting the unsued `_`